R/helpers_outputSummaryFile.R

```diff
@@ -0,0 +1,111 @@
+library(tidyverse)
+
+#' outputSummaryFile
+#' 
+#' Write a summary file to disc. The summary file contains
+#' quality indicators and consists of three sections: 
+#'   - AVERAGE OVER ALL FILES: This section contains the
+#'     pooled standard deviation (O18 and H2) averaged 
+#'     over all files.
+#'     (More parameters will be added in the future.)
+#'   - VALUES FOR EACH FILE: This section contains the 
+#'     pooled standard deviation (O18 and H2) for each file.
+#'   - INTER STANDARD BIAS TO LITERATURE VALUES FOR EACH FILE:
+#'     This section displays the deviation of the measured
+#'     delta.O18 and delta.H2 values to the true values for
+#'     each file.
+#'
+#' @param processedData A list as output by processData(..). 
+#'                      It needs to contain the components
+#'                      $pooledStdDev (a named list where each
+#'                      element is of the form 'list(d18O = .., dD = ..)')
+#'                      and $processed (a named list of dataframes).
+#' @param config A named list. Needs to contain the component 
+#'               $standards (A list of lists. Each innermost list 
+#'               contains the components $name, $o18_True, and  $H2_True).
+#'               Should contain $output_directory if no explicit
+#'               value is given for the argument outputFile.
+#' @param outputFile (Optional) A character vector. Path to the output file.
+#'                   If it is not given, the file 'run.info' in the
+#'                   directory config$output_directory is used.
+#'
+#' @return No relevant return value
+#' 
+outputSummaryFile <- function(processedData, config, outputFile = NULL){
+  
+  # set default value for outputFile if needed
+  if(is.null(outputFile)) 
+    outputFile <- file.path(config$output_directory, "run.info")
```

Comment:
To avoid overwriting issues I would suggest to add a timestamp to the default filename, e.g. `paste(format(Sys.time(),"%Y%m%d%H%M"),"run.info",sep = "_")`

Reply:
Good point, thanks for noticing. 

In the current setup it may actually be better to override existing run.info files. The reason is that doing multiple runs with the same output directory is not supported at the moment. If users did multiple runs with the same output directory the output data (the csv files) would be overriden. Keeping outdated summary files may not provide much added value.